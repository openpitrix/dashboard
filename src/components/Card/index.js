import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

import styles from './index.scss';

export default class Card extends PureComponent {
  static propTypes = {
    icon: PropTypes.string,
    name: PropTypes.string,
    desc: PropTypes.string,
    fold: PropTypes.bool,
  }

  render() {
    const { icon, name, desc, fold } = this.props;
    return (
      <div className={classnames(styles.card, { [styles.foldCard]: fold })}>
        <img src={icon || 'http://via.placeholder.com/96x96'} className={styles.icon} alt="Icon"/>
        <div className={styles.line}></div>
        <p className={styles.name}>{name}</p>
        <p className={styles.desc}>{desc}</p>
      </div>
    );
  }
}
