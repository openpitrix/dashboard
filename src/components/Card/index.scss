@import '~scss/vars';

$card-width: 303px;
$card-height: 208px;

.card {
   float: left;
   display: inline-block;
   position: relative;
   width: $card-width;
   height: $card-height;
   padding: 32px 44px;
   margin: 0 24px 24px 0;
   border-radius: 2px;
   border: 1px solid $N0;
   background-color: $N0;
   transition: all .3s ease-in-out;
   text-align: center;
   box-shadow: 0 1px 4px 0 rgba(73, 33, 173, 0.06), 0 4px 8px 0 rgba(35, 35, 36, 0.04);
   cursor: pointer;
   overflow: hidden;

  .icon {
    width: 48px;
    height: 48px;
    top: 20px;
    margin: 0 auto 12px;
    transition:all 0.3s ease;
    position: relative;
  }
  .name {
    font-family: Roboto sans-serif;
    font-size: 14px;
    color: #343945;
    text-align: center;
    line-height: 28px;
    margin-bottom: 8px;
    transition-delay: 1s;
    transition:all 0.3s ease;
    position: relative;
    left: 0;
    width: 100%;
    height: 28px;
    top: 12px;
  }
  .desc {
    font-family: Roboto-Regular sans-serif;
    font-size: $size-small;
    color: #8C96AD;
    text-align: center;
    line-height: 20px;
    position: relative;
    top: 8px;
    p {
      @include textCut;
    }
  }
  .attrs {
    display: block;
    opacity: 0;
    position: relative;
    bottom: -50px;
    transition:all 0.3s ease;
    dt {
      font-size: 12px;
      color: #8C96AD;
      line-height: 20px;
      display: inline-block;
    }

    dd {
      font-size: 12px;
      color: #576075;
      line-height: 20px;
      display: inline-block;
      margin-left: 0;
    }
    dl {
      margin: 0;
    }
    .label {
      margin-right: 4px;
    }
  }


  &:hover {
    background: $N0;
    border: 1px solid $P45;
    border-position: inside;
    box-shadow: 0 3px 4px 0 rgba(134,145,156,0.04), 0 0 8px 0 rgba(119,133,146,0.04);
    cursor: pointer;
    text-align: left;
    padding: 28px 24px;

    .icon {
      margin: 0;
      vertical-align: middle;
      background-position: left;
      left: -6px;
      transform: scale(0.75);
      float: left;
      top: -6px;
    }

    .name {
      text-align: left;
      left: 2px;
      top: 4px;
      line-height: 28px;
      vertical-align: middle;
      margin-left: 8px;
      height: 28px;
    }
    .desc {
      color: #8C96AD;
      line-height: 20px;
      text-align: left;
      margin-top: 4px;
      margin-bottom: 28px;
      clear: both;
    }
    .attrs {
      bottom: 0;
      opacity: 1;
    }
  }
 }


