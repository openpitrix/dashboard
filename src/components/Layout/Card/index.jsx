import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

import Panel from '../Panel';

import styles from './index.scss';

const Card = ({
  className, children, hasTable = false, ...others
}) => (
  <Panel
    className={classnames(styles.card, className, {
      [styles.tableCard]: hasTable
    })}
    {...others}
  >
    {children}
  </Panel>
);

Card.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  hasTable: PropTypes.bool
};

export default Card;
