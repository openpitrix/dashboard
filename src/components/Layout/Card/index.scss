$card-padding: 24px;

.card {
  position: relative;
  box-sizing: border-box;
  padding: $card-padding;
  margin-bottom: 24px;

  :global {
    .operation{
      position: absolute;
      right: 32px;
      top: 24px;
      cursor: pointer;
      .pi-popover-popper {
        margin-left: -90px !important;
      }
    }
  }
}

.tableCard {
  padding: $card-padding;

  :global {
    .loadTable {
      position: sticky !important;
      margin-top: 0;
    }
  }
}
