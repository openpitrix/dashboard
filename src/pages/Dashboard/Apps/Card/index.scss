@import '~scss/vars';
.container {
  @include note-font;
  padding: 32px;
  background: $N0;
  box-shadow: 0 1px 4px 0 rgba($P400, 0.06), 0 4px 8px 0 rgba(35, 35, 36, 0.04);
  width: calc(25% - 20px);
  margin: 0 10px 20px;
  cursor: pointer;
  .status {
    height: inherit;
  }
}

@media screen and (min-width: 2468px) {
  /* .cards { */
  /*   grid-template-columns: repeat(6, 1fr); */
  /* } */
  .container {
    width: calc(16.66% - 20px);
  }
}

.title {
  display: flex;
}

.name {
  @include title-font;
  @include textCut;
  max-width: 148px;
}

.imageOuter {
  display: inline-block;
  margin-right: 8px;
  width: 48px;
  height: 48px;
  line-height: 48px;
}

.description {
  @include note-font;
  display: -webkit-box;
  /* autoprefixer: off */
  -webkit-box-orient: vertical;
  /* autoprefixer: on */
  -webkit-line-clamp: 3;

  margin: 20px 0;
  height: 60px;
  width: 100%;
  overflow: hidden;
}

.deliverTypes {
  margin-bottom: 8px;
}
.deliverType {
  margin-right: 12px;
  color: $N300;
}
