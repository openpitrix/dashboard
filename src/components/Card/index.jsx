import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import { withRouter } from 'react-router';

import { Icon, Image } from 'components/Base';

import styles from './index.scss';

class Card extends PureComponent {
  static propTypes = {
    desc: PropTypes.string,
    fold: PropTypes.bool,
    icon: PropTypes.string,
    link: PropTypes.string,
    name: PropTypes.string
  };

  handleClick = () => {
    const { history, link } = this.props;

    history.push(link);
  };

  render() {
    const {
      icon, name, desc, fold
    } = this.props;
    const iconSize = fold ? 36 : 48;

    return (
      <div
        className={classnames(styles.card, { [styles.foldCard]: fold })}
        onClick={this.handleClick}
      >
        <div className={styles.title}>
          <span className={styles.icon}>
            <Image
              src={icon}
              alt="Icon"
              iconSize={iconSize}
              iconLetter={name}
            />
          </span>
          <p className={styles.name}>{name}</p>
          <p
            className={classnames(styles.desc, { [styles.hide]: !fold })}
            title={desc}
          >
            {desc}
          </p>
        </div>
        <div className={classnames(styles.line, { [styles.hide]: fold })} />
        <p
          className={classnames(styles.desc, { [styles.hide]: fold })}
          title={desc}
        >
          {desc}
        </p>
      </div>
    );
  }
}

export default withRouter(Card);
