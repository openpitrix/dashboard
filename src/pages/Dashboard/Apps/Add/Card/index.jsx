import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { observer } from 'mobx-react';
import classNames from 'classnames';
import { withTranslation } from 'react-i18next';

import { Icon, DocLink } from 'components/Base';
import externalLink from 'config/external-link';

import styles from './index.scss';

@withTranslation()
@observer
export default class CreateAppCard extends Component {
  static propTypes = {
    className: PropTypes.string,
    intro: PropTypes.string,
    name: PropTypes.string,
    value: PropTypes.string
  };

  render() {
    const {
      name,
      value,
      icon,
      intro,
      className,
      disable,
      t,
      appCreateStore
    } = this.props;
    const {
      selectVersionType,
      checkAddedVersionType,
      checkSelectedVersionType
    } = appCreateStore;
    const isAdded = checkAddedVersionType(value);
    const isSelected = checkSelectedVersionType(value);

    return (
      <div
        onClick={() => !isAdded && !disable && selectVersionType(value)}
        className={classNames(styles.container, className, {
          [styles.addedContainer]: isAdded,
          [styles.selected]: isSelected,
          [styles.disabledContainer]: disable
        })}
      >
        {isAdded && <span className={styles.addedType}>{t('Added')} </span>}
        {isSelected && (
          <Icon
            className={styles.checkedIcon}
            name="checked-circle"
            size={20}
          />
        )}
        <Icon name={icon} size={48} className={styles.icon} type="light" />
        <div className={styles.name}>{name}</div>
        <div className={styles.intro}>{t(intro)}</div>
        <DocLink
          name={'Create_App_Linkto_Intro_App'}
          to={externalLink[`doc_${value}`]}
        >
          {t('Linkto_Intro_App')}
        </DocLink>
      </div>
    );
  }
}
