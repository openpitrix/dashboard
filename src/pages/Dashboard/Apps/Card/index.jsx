import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import { withRouter } from 'react-router';

import { Image } from 'components/Base';
import Status from 'components/Status';
import { getPastTime } from 'src/utils';
import { getVersionTypesName } from 'config/version-types';

import styles from './index.scss';

@withRouter
export default class AppCard extends Component {
  static propTypes = {
    children: PropTypes.node,
    className: PropTypes.string,
    data: PropTypes.object
  };

  render() {
    const { className, data, t } = this.props;
    const {
      app_id,
      icon,
      name,
      description,
      status,
      status_time,
      app_version_types
    } = data;
    return (
      <div
        className={classnames(styles.container, className)}
        onClick={() => {
          this.props.history.push(`/dashboard/app/${app_id}/versions`);
        }}
      >
        <div className={styles.title}>
          <label className={styles.imageOuter}>
            <Image iconLetter={name} src={icon} iconSize={48} alt="logo" />
          </label>
          <div>
            <p className={styles.name}>{name}</p>
            <Status className={styles.status} name={status} type={status} />
          </div>
        </div>
        <pre className={styles.description}>{description}</pre>
        <div className={styles.deliverTypes}>
          <span>{t('Delivery type')}：</span>
          <span className={styles.deliverType}>
            {(getVersionTypesName(app_version_types) || []).join(' ')}
          </span>
        </div>
        <div>
          <span>{t('Updated At')}：</span>
          <span>{getPastTime(status_time)}</span>
        </div>
      </div>
    );
  }
}
