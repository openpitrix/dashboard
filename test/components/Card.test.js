import { BrowserRouter } from 'react-router-dom';
import Card from 'components/Card';

describe('Card', () => {
  it('basic render', () => {
    const wrapper = render(
      <BrowserRouter>
        <Card />
      </BrowserRouter>
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
