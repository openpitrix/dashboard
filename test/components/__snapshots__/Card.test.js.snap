// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card basic render 1`] = `
<div
  class="card"
>
  <div
    class="icon"
  >
    <img
      alt="Icon"
      class="img"
      data-origin-url=""
      src=""
    />
  </div>
  <div
    class="name"
  />
  <pre
    class="desc"
  />
  <div
    class="attrs"
  >
    <dl>
      <dt
        class="label"
      >
        Service provider:
      </dt>
      <dd
        class="val"
      >
        unknown
      </dd>
    </dl>
    <dl>
      <dt
        class="label"
      >
        Delivery type:
      </dt>
      <dd
        class="val"
      >
        <div
          class="versionType"
        />
      </dd>
    </dl>
  </div>
</div>
`;
